delivery/repositories.py

```diff
@@ -2,48 +2,93 @@
     def __init__(self, restaurants=None):
         self.restaurants = restaurants or []
 
-    def create(self, content: Restaurant) -> None:
-        content.id = len(self.restaurants)
+    def create(self, content: Restaurant) -> Restaurant:
+        content.id = str(len(self.restaurants))
         self.restaurants.append(content)
+        return content
 
-    def update(self, content: Restaurant) -> None:
+    def update(self, content: Restaurant) -> Restaurant:
         for restaurant in self.restaurants:
             if restaurant.id == content.id:
                 for key in ["name", "address", "work_time", "phone_number"]:
                     if getattr(content, key) != "":
                         setattr(restaurant, key, getattr(content, key))
-                return
+                return restaurant
         raise WrongIdError(description="Restaurant with such ID doesn't exist")
 
     def get_all(self) -> Iterable[Restaurant]:
         return self.restaurants
 
-    def get_by_id(self, restaurant_id: int) -> Restaurant:
+    def get_by_id(self, restaurant_id: str) -> Restaurant:
         for restaurant in self.restaurants:
             if restaurant.id == restaurant_id:
                 return restaurant
         raise WrongIdError(description="Restaurant with such ID doesn't exist")
+
+
+class MongoRestaurantRepository(AbstractRestaurantRepository):
+    def __init__(self, mongo_client: MongoClient):
+        self.mongo_client = mongo_client.restaurants
+
+    def create(self, content: Restaurant) -> Restaurant:
+        content = asdict(content)
+        content.pop("id")
+        return self.mongo_client.find_one_and_update({'name': ''},
+                                                     {'$set': content},
+                                                     upsert=True, return_document=ReturnDocument.AFTER)
+
+    def update(self, content: Restaurant) -> Restaurant:
+        update_data = asdict(content)
+        restaurant_id = {'_id': ObjectId(update_data["id"])}
+        for key, value in asdict(content).items():
+            if value == "" or key == "id":
+                update_data.pop(key)
+        restaurant = self.mongo_client.find_one_and_update(restaurant_id, {"$set": update_data}, return_document=ReturnDocument.AFTER)
+        if restaurant is None:
+            raise WrongIdError(description="Restaurant with such ID doesn't exist")
+        restaurant["id"] = restaurant.pop("_id")
+        return Restaurant(**restaurant)
+
+    def get_all(self) -> Iterable[Restaurant]:
+        restaurants_list = []
+        for restaurant in self.mongo_client.find({}):
+            restaurant["id"] = restaurant.pop("_id")
+            restaurants_list.append(Restaurant(**restaurant))
+        return restaurants_list
+
+    def get_by_id(self, restaurant_id) -> Restaurant:
+        restaurant = self.mongo_client.find_one({"_id": ObjectId(restaurant_id)})
+        try:
```

Comment:
`find_one` will return `None` if the object is not found, so it should be 
```python
if restaurant is None:
    raise WrongIdError(description="Restaurant with such ID doesn't exist")
```

Reply:
fixed